sql/sql_expression_operations.cc

```diff
@@ -814,6 +814,7 @@ struct BinaryOpHelper {
                                 std::placeholders::_1,
                                 std::placeholders::_2,
                                 GET_LATEST);
+        result.expr = expr->shared_from_this();
```

Comment:
I believe that the pattern is OK for expression trees; otherwise any tree manipulation will need to pass std::shared_ptr<SqlExpression> thisIsASharedPointerToMe (which is what it was early on, and believe me the current way is a lot better).  What we really need to do is to make it impossible to allocate an SqlExpression apart from via std::make_shared<>.